src/detect-lua.c

```diff
@@ -687,6 +687,7 @@ static DetectLuaData *DetectLuaParse (const DetectEngineCtx *de_ctx, const char
         goto error;
     }
 
+    lua->de_ctx = (DetectEngineCtx *)de_ctx;
```

Comment:
I suppose this is needed as we don't pass `de_ctx` to the free logic. Would be nice if we can change that to avoid hacks like this.